sdk/communication/AzureCommunication/Source/Authentication/CommunicationTokenCredential.swift

```diff
@@ -63,12 +87,38 @@ public typealias TokenRefreshHandler = (String?, Error?) -> Void
         )
     }
 
+    /**
+     Creates a CommunicationTokenCredential that automatically refreshes the token.
+     - Parameter delegate: Delegate class conforming to `TokenCredentialDelegate`
+     - Parameter option: Options for how the token will be refreshed
+     - Throws: `AzureError` if the provided token is not a valid user token.
+     */
+    public init(
```

Comment:
@sacheu  @tjprescott I wonder does this make sense to have an init with with delegate when using options? 
It seems like an odd contradiction. The options object itself has a `tokenRefresher` closure we pass in, but if they are using the delegate pattern they would expect us to call the new `onTokenRetrieved` method from the delegate. 

So does it make more sense to create a new init looking like this 
```swift 
public init(withDelegate delegate: TokenCredentialDelegate, refreshProactively: Bool, initialToken: String) throws
```

Reply:
Are there other options? I think the approach you suggest here makes sense, so long as any additional options are still available to the delegate.

Reply:
One simple solution could be to make a change `var tokenRefresher: TokenRefresher?`

<img width="588" alt="Screen Shot 2021-04-15 at 1 12 04 PM" src="https://user-images.githubusercontent.com/9044372/114932333-36132880-9dec-11eb-908f-aeaa46e65f9a.png">

Or we can create a new options object where there is no closure 
```swift
@objcMembers public class CommunicationTokenRefreshSimpleOptions: NSObject {
    var initialToken: String?
    var refreshProactively: Bool
}
```

Reply:
This is very similar to what we did in C#:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Common/src/CommunicationTokenRefreshOptions.cs